### ISSUE TITLE
KLayout will crash if re-run Macro with GUI 

### ISSUE THREAD

**Issue by zwh42** [KLayout/klayout#298]
Hi all,

I've wrote a Python macro with a simple GUI to convert to image to Oasis layout file https://github.com/zwh42/KLayout-PyMacros/tree/master/image_to_oasis_file . During the development, everytime I close the marcro and re-run the macro from the macro development, the whole KLayout will crash. Could you please help on this?  My KLayout version is 0.25.8.

Thanks!

**Comment by klayoutmatthias** [comment]
Hi,

Maybe you can try the 0.26 beta builds from "www.klayout.org/downloads/master". Some stability issues have been fixed there. There is also 0.25.9 which got a few fixes that might be helpful.

What's the system you're on?

I'll try whether I can reproduce the issue with your script. I have Linux and Windows.

Matthias

**Comment by zwh42** [comment]
Thanks Matthias, both MacOS(10.14.5) and Windows 10 have this issue.
I will try the new update.

**Comment by klayoutmatthias** [comment]
Good news is I can reproduce it with 0.25, but with 0.26 I can’t 😊

It wasn’t working for initially because of a small issue in the code. I created a pull request.

I did not do a full analysis, but I reason the problem was because of “show” as the last statement. When the macro is rerun, the dialog object is deleted. But then the window might still be shown. In this case, Qt has to remove the Window first before the object is deleted. I recall an issue with situations like this and I guess it has been fixed.

In general modal dialogs (exec instead of show) are easier to manage and there is no interference of this kind. You might also consider keeping the dialog object (not creating it if is already there). This way, users will also find their window again when they rerun the macro.

Best regards,

Matthias


P.S: here is how it looks for me on W10:
